layers/+distribution/spacemacs-base/config.el

```diff
@@ -226,7 +226,20 @@ nil."
   (add-hook 'prog-mode-hook 'linum-mode)
   (add-hook 'text-mode-hook 'linum-mode))
 ;; line number
-(setq linum-format "%4d")
+;; add a space before/after the line number
+;; and make number width more flexible to suit with the max lines
+(defun linum-format-func (line)
+  (concat
+    (propertize " " 'face 'linum)
+    (propertize (format linum-format-fmt line) 'face 'linum)
+    (propertize " " 'face 'linum)))
+(add-hook 'linum-before-numbering-hook
+  (lambda ()
+    (setq-local linum-format-fmt
+    (let ((w (length (number-to-string
+          (count-lines (point-min) (point-max))))))
```

Comment:
Not sure about the efficient of this function, it should be profiled against large buffers (in number of lines).